File: AzureGateway.UI/src/models/GpsServerSetting.ts
export interface GpsServerSetting {
    id: number;
    isEnabled: boolean;
    dataFolderPath: string;
    updatedAt?: string;
    updatedBy?: string | null;
}

export interface UpdateGpsServerSettingRequest {
    isEnabled: boolean;
    dataFolderPath: string;
}

File: MonitoringServiceAPI/MonitoringServiceAPI/Models/GpsServerSetting.cs
using System.ComponentModel.DataAnnotations;

namespace MonitoringServiceAPI.Models
{
    public class GpsServerSetting
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public bool IsEnabled { get; set; } = false;

        [Required]
        [StringLength(500)]
        public string DataFolderPath { get; set; } = string.Empty;
    }
}

